pkg/config/update.go

```diff
@@ -0,0 +1,236 @@
+		writeSet[groupName] = groupWriteSet
+
+	}
+
+	for groupName, updatedGroup := range updated {
+		if _, ok := original[groupName]; ok {
+			// If the updatedGroup is in the original set of groups, it was already handled
+			continue
+		}
+		updatedMembers = true
+		_, groupWriteSet, _ := computeGroupUpdate(newConfigGroup(), updatedGroup)
+		writeSet[groupName] = &cb.ConfigGroup{
+			Version:   0,
+			ModPolicy: updatedGroup.ModPolicy,
+			Policies:  groupWriteSet.Policies,
+			Values:    groupWriteSet.Values,
+			Groups:    groupWriteSet.Groups,
+		}
+	}
+
+	return
+}
+
+func computeGroupUpdate(original, updated *cb.ConfigGroup) (readSet, writeSet *cb.ConfigGroup, updatedGroup bool) {
+	readSetPolicies, writeSetPolicies, sameSetPolicies, policiesMembersUpdated := computePoliciesMapUpdate(original.Policies, updated.Policies)
+	readSetValues, writeSetValues, sameSetValues, valuesMembersUpdated := computeValuesMapUpdate(original.Values, updated.Values)
+	readSetGroups, writeSetGroups, sameSetGroups, groupsMembersUpdated := computeGroupsMapUpdate(original.Groups, updated.Groups)
+
+	// If the updated group is 'Equal' to the updated group (none of the members nor the mod policy changed)
+	if !(policiesMembersUpdated || valuesMembersUpdated || groupsMembersUpdated || original.ModPolicy != updated.ModPolicy) {
+
+		// If there were no modified entries in any of the policies/values/groups maps
+		if len(readSetPolicies) == 0 &&
+			len(writeSetPolicies) == 0 &&
+			len(readSetValues) == 0 &&
+			len(writeSetValues) == 0 &&
+			len(readSetGroups) == 0 &&
+			len(writeSetGroups) == 0 {
+			return &cb.ConfigGroup{
+					Version: original.Version,
+				}, &cb.ConfigGroup{
+					Version: original.Version,
+				}, false
+		}
+
+		return &cb.ConfigGroup{
+				Version:  original.Version,
+				Policies: readSetPolicies,
+				Values:   readSetValues,
+				Groups:   readSetGroups,
+			}, &cb.ConfigGroup{
+				Version:  original.Version,
+				Policies: writeSetPolicies,
+				Values:   writeSetValues,
+				Groups:   writeSetGroups,
+			}, true
+	}
+
+	for k, samePolicy := range sameSetPolicies {
+		readSetPolicies[k] = samePolicy
+		writeSetPolicies[k] = samePolicy
+	}
+
+	for k, sameValue := range sameSetValues {
+		readSetValues[k] = sameValue
+		writeSetValues[k] = sameValue
+	}
+
+	for k, sameGroup := range sameSetGroups {
+		readSetGroups[k] = sameGroup
+		writeSetGroups[k] = sameGroup
+	}
+
+	return &cb.ConfigGroup{
+			Version:  original.Version,
+			Policies: readSetPolicies,
+			Values:   readSetValues,
+			Groups:   readSetGroups,
+		}, &cb.ConfigGroup{
+			Version:   original.Version + 1,
+			Policies:  writeSetPolicies,
+			Values:    writeSetValues,
+			Groups:    writeSetGroups,
+			ModPolicy: updated.ModPolicy,
+		}, true
+}
+
+// Compute computes the difference between two *cb.Configs and returns the
+// ReadSet and WriteSet diff as a *cb.ConfigUpdate
+func Compute(original, updated *cb.Config) (*cb.ConfigUpdate, error) {
```

Comment:
nit: feels weird to have the only exported function in this file at the very end and not the beginning.

Reply:
fixed 🔧